### DISCUSSION TITLE
Is there a way for us to override/update the underlying Alpine ISO for the VM to compile a kernel with CONFIG_FANOTIFY=Y?

### DISCUSSION THREAD

**Discussion by simensen** [abiosoft/colima#633] [Q&A] [answered]
One of my services would benefit from using fanotify. The software will use it if it is detected. I believe my cotainer images are set up to leverage fanotify (and have set `CAP_SYS_ADMIN` when the containers are created) but it seems like the VM image wasn't built with `CONFIG_FANOFITY=Y` itself, so the containers can't leverage it.

I'm not 100% sure how the colima alpine fork is supposed to work, but it looks like aports is pinned at 3d0a466 from ~a year ago.

Would it be possible to build the ISO with support for fanotify ourselves? If not, what's the best path to getting this built colima's images in the future?

**Comment by abiosoft** [comment, marked as answer]
Any idea if it is something that can be enabled via kernel flag? e.g. https://github.com/abiosoft/colima/issues/384#issuecomment-1235753254

**Reply by jandubois** [reply]
> Any idea if it is something that can be enabled via kernel flag? e.g. [#384 (comment)](https://github.com/abiosoft/colima/issues/384#issuecomment-1235753254)

It needs to be configured when the kernel is compiled. I believe it is included in Alpine 3.17 but not in 3.16, but have not tested that it actually works in 3.17.

**Reply by abiosoft** [reply]
Unfortunately 3.17 is problematic at the moment and I do not have the time luxury to troubleshoot why it's broken for Lima.

**Reply by jandubois** [reply]
I'm aware that it doesn't work with VZ on aarch64. Please let me know if there are any other issue with Alpine 3.17 and Lima 0.15, as I'm not aware of any (and therefore wouldn't look into it).

**Reply by simensen** [reply]
Thanks. I'm pretty new to doing anything this low-level with Lima/Colima, so just knowing what you've shared here has been a huge help. Namely: there isn't anything easy I can do to get this working myself to try it out.

I realize I left out my use case. I'm doing more work with https://mutagen.io/ and their agent can leverage fanotify if the target (alpha or beta) are configured to enable fanotify. Since the functionality wasn't kicking in, I wanted to see if it was a problem at the docker/container level or if it was the underlying VM docker was running on.
